src/zenml/orchestrators/step_launcher.py

```diff
@@ -135,8 +138,8 @@ def __init__(
         self._deployment = deployment
         self._step = step
         self._orchestrator_run_id = orchestrator_run_id
-        stack_model = Client().get_stack(deployment.stack_id)
-        self._stack = Stack.from_model(stack_model)
+        assert deployment.stack
```

Comment:
I imagine deleting a stack could result in this, should we raise a proper error message then perhaps? Or maybe make stack non-optional and have cascade delete on the stack-deployment relationship?

Reply:
Similar to my comment above, at that point they're just messing with their runs and there's not much we can do. If the stack that the current pipeline is running on doesn't exist anymore, everything will fail and we probably won't even get to this point